libcxx/include/string

```diff
@@ -1948,8 +1948,8 @@ private:
     if (__s < __min_cap) {
       return static_cast<size_type>(__min_cap) - 1;
     }
-    size_type __guess =
-        __align_it < sizeof(value_type) < __alignment ? __alignment / sizeof(value_type) : 1 > (__s + 1) - 1;
+    const size_type __boundary = sizeof(value_type) < __alignment ? __alignment / sizeof(value_type) : 1;
```

Comment:
Alternatively, we could also make `__boundary` part of the enumerator?

I have a weak distaste for using `const` on a variable we need to be `constexpr`. But  it shouldn't be an issue here at all.

Reply:
I agree it's annoying to use `const` when we mean `constexpr`. I'm not certain about moving to the enumerator since this feels more like a temporary variable (an implementation detail) of the function, whereas the alignment is a more visible "configurable knob". Also, I'm tempted to merge as-is to avoid merge conflicts with your other patch that touches the eumerator :-).